### PR TITLE
Create generate_report_ui.gs

### PR DESCRIPTION
**Pull request by siddhimahawar**
This is the UI for the generate report card of Drive add-on

### CODE REVIEW

File: generate_report_ui.gs [hunk 1]
```diff
@@ -0,0 +1,263 @@
+/**
+ * Callback for rendering the view.
+ * @return {CardService.Card} The card to show to the user.
+ */
+function onHomePage(e) {
```
Comment (shikhagupta1212) [inline generate_report_ui.gs:5]: Parameter must have JSDoc. Please fix here and below

File: generate_report_ui.gs [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    timeFilter.addItem("Last 7 Days", "LAST_7_DAYS", e.formInput.timeFilter == "LAST_7_DAYS");
+    timeFilter.addItem("Last 30 Days", "LAST_30_DAYS", e.formInput.timeFilter == "LAST_30_DAYS");
+  }
+  else {
+    timeFilter.addItem("Last Hour", "LAST_HOUR", true);
+    timeFilter.addItem("Last 6 Hours", "LAST_6_HOUR", false);
+    timeFilter.addItem("Last 24 Hours", "LAST_24_HOUR", false);
+    timeFilter.addItem("Last 7 Days", "LAST_7_DAYS", false);
+    timeFilter.addItem("Last 30 Days", "LAST_30_DAYS", false);
+  }
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: This code is duplicated. Can you create a function in which may be you can pass a map of filters you want to apply, selectionInput and default value.

File: generate_report_ui.gs [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+function generateReport(e) {
+  var title, projTitle;
+  var toTime = new Date();
+  var fromTime;
+  var millisInBetween;
+  var input  = e.formInput;
+  var projectFilter = input.projectFilter;
+  var reportType = input.reportType;
+  var limitResponse = input.limitResponse;
+  if(!limitResponse) limitResponse = 1;
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: nit: use {} around if

File: generate_report_ui.gs [hunk 4]
```diff
@@ -0,0 +1,263 @@
+/**
+ * Callback for rendering the view.
+ * @return {CardService.Card} The card to show to the user.
+ */
+function onHomePage(e) {
+  return createUI(e);
```
Comment (shikhagupta1212) [inline generate_report_ui.gs:6]: Parameter must have JSDoc. Please fix here and below

File: generate_report_ui.gs [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+  var fromTime;
+  var millisInBetween;
+  var input  = e.formInput;
+  var projectFilter = input.projectFilter;
+  var reportType = input.reportType;
+  var limitResponse = input.limitResponse;
+  if(!limitResponse) limitResponse = 1;
+    
+  if(input.timeFilter == "LAST_HOUR") { 
+    title = "Last Hour";
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: rename title to timestamp_header or similar to it

File: generate_report_ui.gs [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+  } else if(projectFilter == "CUSTOM_PROJECT") {
+    projTitle = "Custom Projects";
+  } else if(projectFilter == "SYSTEM_PROJECT") {
+    projTitle = "System Projects";
+  } else {
+    projTitle = "All Projects";
+  }
+  
+  // variables to get the start time and end time
+  var fromTime = new Date(toTime.getTime() - millisInBetween);
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: In the same function there are too many declaration of fromTime. Please fix that. One variable should be declared once. Similarly below

File: generate_report_ui.gs [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+  // if the selected option is specific project
+  if(projectFilter == "SPECIFIC_PROJECT") {
+    if(reportType == "mostExecutedScript") {
+      var processes = getNumberOfExecutionOfScript(input.projectId, fromTime);
+      var mostExecutedScript = convertObjectToSortedArrayForSpecificProject(processes, input.projectId);
+      var reportUrl = generateReportForMostExecutedScript(mostExecutedScript, "Report for Most Executed Script of "+ projTitle + " for " + title, limitResponse);
+    }
+    else if(reportType == "mostActiveUsers") {
+      var users = getUsersWithProcessId(input.projectId, fromTime);
+      var mostActiveUsers = convertObjectToSortedArrayForSPECIFIC_PROJECT(users, input.projectId);
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: Fix the naming convention. Naming should be in camel case only

File: generate_report_ui.gs [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+  var fromTime = Utilities.formatDate(fromTime, 'Etc/GMT', 'yyyy-MM-dd\'T\'HH:mm:ss.SSS\'Z\'');
+  
+  // if the selected option is specific project
+  if(projectFilter == "SPECIFIC_PROJECT") {
+    if(reportType == "mostExecutedScript") {
+      var processes = getNumberOfExecutionOfScript(input.projectId, fromTime);
+      var mostExecutedScript = convertObjectToSortedArrayForSpecificProject(processes, input.projectId);
+      var reportUrl = generateReportForMostExecutedScript(mostExecutedScript, "Report for Most Executed Script of "+ projTitle + " for " + title, limitResponse);
+    }
+    else if(reportType == "mostActiveUsers") {
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: nit: else if the same line as }

File: generate_report_ui.gs [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+    if(reportType == "mostExecutedScript") {
+      var mostExecutedScript = getMostExecutedScriptFromAllCloudProjects(fromTime , projectFilter);
+      var reportUrl = generateReportForMostExecutedScript(mostExecutedScript, "Report for Most Executed Script of "+ projTitle + " for " + title, limitResponse);
+    }
+    else if(reportType == "mostActiveUsers") {
+      var mostActiveUser = getMostActiveUser(fromTime , projectFilter);
+      var reportUrl = generateReportForMostActiveUsers(mostActiveUser, "Report for Most Active Users of "+ projTitle + " for " + title, limitResponse);
+    }
+    else
+    {
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: { braces has to be in same line as else or function

File: generate_report_ui.gs [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+      .setNavigation(navigation);
+    return actionResponse.build();
+}
+
+/**
+* Callback for the project Filter
+* @return {CardService.Card} The card to show to the user
+*/
+function projectFilterCallback(e){
+
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: nit: add extra space b/w ){

File: generate_report_ui.gs [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+  var card = createUI(e);
+    var navigation = CardService.newNavigation()
+      .updateCard(card);
+    var actionResponse = CardService.newActionResponseBuilder()
+      .setNavigation(navigation);
+    return actionResponse.build();
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: Fix indentation

File: generate_report_ui.gs [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+* @return {CardService.Card} The card to show to the user
+*/
+function reportTypeCallback(e){
+
+  var card = createUI(e);
+    var navigation = CardService.newNavigation()
+      .updateCard(card);
+    var actionResponse = CardService.newActionResponseBuilder()
+      .setNavigation(navigation);
+    return actionResponse.build();
```
Comment (shikhagupta1212) [inline generate_report_ui.gs]: Same as above

File: generate_report_ui.gs (hunk 2 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Card Service has certain limitations.
I tried doing it in function but wasn't successful

File: generate_report_ui.gs (hunk 3 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

File: generate_report_ui.gs (hunk 5 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

File: generate_report_ui.gs (hunk 6 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

File: generate_report_ui.gs (hunk 7 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

File: generate_report_ui.gs (hunk 8 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

File: generate_report_ui.gs (hunk 9 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

File: generate_report_ui.gs (hunk 10 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

File: generate_report_ui.gs (hunk 11 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

File: generate_report_ui.gs (hunk 12 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

File: generate_report_ui.gs (hunk 4 above)
Comment (siddhimahawar) [reply generate_report_ui.gs:6]: Updated

File: generate_report_ui.gs (hunk 2 above)
Comment (siddhimahawar) [reply generate_report_ui.gs]: Updated

**Review by piy314 — APPROVED** [review]

**Pull request merged**